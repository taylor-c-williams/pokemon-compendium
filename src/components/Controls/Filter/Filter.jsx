export default function Filter({ allTypes, selectedType, setSelectedType }) {
	return (
		<div className="filter-body">
			{/* Type Selector */}
			<select
				aria-label="typeFilter"
				value={selectedType}
				onChange={(e) => setSelectedType(e.target.value)}
			>
				<option key="all" value="all">
					All
				</option>
				{allTypes.map((type) => {
					return (
						<option key={type} value={type}>
							{type}
						</option>
					);
				})}
			</select>
		</div>
	);
}
