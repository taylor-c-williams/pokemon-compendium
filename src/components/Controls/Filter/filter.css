.filter-body {
	background-image: url('../clouds.png');
	background-repeat: no-repeat;
	background-size: contain;
}
