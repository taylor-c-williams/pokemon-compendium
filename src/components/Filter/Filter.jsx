import React from 'react';

export default function Filter({
	allTypes,
	selectedType,
	sortOrder,
	handleChange,
	handleSort,
}) {
	return (
		<div>
			{/* Type Selector */}
			<select
				value={selectedType}
				onChange={(e) => handleChange(e.target.value)}
			>
				<option key="all" value="all">
					All
				</option>
				{allTypes.map((type) => {
					return (
						<option key={type} value={type}>
							{type}
						</option>
					);
				})}
			</select>

			{/* Sort Order */}
			<select value={sortOrder} onChange={(e) => handleSort(e.target.value)}>
				<option value="asc">Ascending</option>
				<option value="desc">Descending</option>
			</select>
		</div>
	);
}
