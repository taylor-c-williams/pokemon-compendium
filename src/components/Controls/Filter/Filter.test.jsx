import { screen, render } from '@testing-library/react';
import Filter from './Filter';

it('should properly render the Filter component', () => {
	const allTypes = ['normal', 'bug', 'water', 'electric', 'ghost'];
	render(<Filter allTypes={allTypes} />);
	const typeFilter = screen.getAllByLabelText('typeFilter');
	expect(typeFilter).toMatchSnapshot();
});
