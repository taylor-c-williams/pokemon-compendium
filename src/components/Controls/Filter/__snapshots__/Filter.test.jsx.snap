// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should properly render the Filter component 1`] = `
Array [
  <select
    aria-label="typeFilter"
  >
    <option
      value="all"
    >
      All
    </option>
    <option
      value="normal"
    >
      normal
    </option>
    <option
      value="bug"
    >
      bug
    </option>
    <option
      value="water"
    >
      water
    </option>
    <option
      value="electric"
    >
      electric
    </option>
    <option
      value="ghost"
    >
      ghost
    </option>
  </select>,
]
`;
